File: app_watcher/src/main/java/com/tongchen/twatcher/base/ui/activity/BaseActivity.java
package com.tongchen.twatcher.base.ui.activity;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;


public abstract class BaseActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        /*if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            Window window = getWindow();
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS | WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.setStatusBarColor(getResources().getColor(R.color.colorAccent));
        }*/

        setContentView(bindLayout());
        //  绑定ButterKnife
        ButterKnife.bind(this);
        loadView();

    }

    @LayoutRes
    protected abstract int bindLayout();

    protected abstract void loadView();

}

File: app_watcher/src/main/java/com/tongchen/twatcher/base/ui/activity/BaseActivity.kt
package com.tongchen.twatcher.base.ui.activity

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity

/**
 * @author TongChen
 * @date 2019/11/29  18:20
 * <p>
 * Desc:
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        /*if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS or WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            window.statusBarColor = resources.getColor(R.color.colorAccent)
        }*/

        setContentView(bindLayout())
        loadView()
    }

    @LayoutRes
    protected abstract fun bindLayout(): Int

    protected abstract fun loadView()
}